store/src/java/com/zimbra/cs/service/util/JWTUtil.java

```diff
@@ -179,15 +183,18 @@ private static String getJWTSalt(String jwt, String salts) throws ServiceExcepti
             String[] saltArr = salts.split("\\|");
             byte[] tokenKey = getTokenKey();
             for (String salt:saltArr) {
-                byte[] finalKey = Bytes.concat(tokenKey, salt.getBytes());
-                Key key = new SecretKeySpec(finalKey, SignatureAlgorithm.HS512.getJcaName());
-                try {
-                    Claims claims = Jwts.parser().setSigningKey(key).parseClaimsJws(jwt).getBody();
-                    jwtSpecificSalt = salt;
-                    JWTCache.put(jti, new ZimbraJWT(salt, claims.getExpiration().getTime()));
-                    break;
-                } catch(Exception e) {
-                    //invalid salt, continue to try another salt value
+                if (!StringUtil.isNullOrEmpty(salt)) {
+                    byte[] finalKey = Bytes.concat(tokenKey, salt.getBytes());
+                    Key key = new SecretKeySpec(finalKey, SignatureAlgorithm.HS512.getJcaName());
+                    try {
+                        Claims claims = Jwts.parser().setSigningKey(key).parseClaimsJws(jwt).getBody();
+                        jwtSpecificSalt = salt;
+                        JWTCache.put(jti, new ZimbraJWT(salt, claims.getExpiration().getTime()));
+                        break;
+                    } catch(Exception e) {
+                        //invalid salt, continue to try another salt value
+                        ZimbraLog.account.debug("invalid salt, continue to try another salt value", e);
```

Comment:
Make sure we are not logging any token or JTI in mailbox.log  This will be a security flaw. 

For any sensitive information use zimbra.security logger which add into audit.log